### DISCUSSION TITLE
Is PactSwift still very active?

### DISCUSSION THREAD

**Discussion by perlguy99** [surpher/PactSwift#109] [Q&A]
First off, I'd like to thank everyone who contributed on this project, it is very nice!

I am finding that trying to find information on PactSwift on the internet is pretty scarce, is PactSwift the only option for iOS?

Also, does anyone know why the PactSwiftMockServer ends up being almost 1Gb in size when built?

Thank you!

**Comment by surpher** [comment]
Hi @perlguy99 

Thanks. PactSwift is still active (eg: https://github.com/surpher/PactSwift/tree/feat/v2.0.0) and I'm keeping a half-shut eye out on things happening here and in Slack. The problem I have is I'm quite time poor and most of my `git` and repo interaction with current client is on Azure DevOps meaning I don't venture over here as much as I'd like.

Whether PactSwift is the only option for iOS, for Pact Specification v3 (and v4 the branch above), afaik, yes. 
There is a sunsetted `pact-consumer-swift` out there too which I was personally involved with quite a bit, but it's only v2 and a few more steps are needed to set it up. It's just deprecated now.

And why `PactSwiftMockServer` is so big:  https://github.com/surpher/PactSwift/issues/117 
There was an attempt to get users of this framework to do a bit of jumping through hoops, have `rust` installed on their devices, on their CI agents etc and... It was just not worth it compared to eating up the HDD space. 

I had a plan to clean up the old binaries out of the repo to reduce its' size but am time poor right now. Happy to guide anyone else wanting to have a crack at it.

**Comment by surpher** [comment]
I actually have an idea how to avoid pulling in the dev part of `PactSwiftMockServer` so it doesn't pull in all the unnecessary history. Bare with me for a couple of days!

edit:
@perlguy99 Try pointing your project's dependency to `main` branch of `PactSwift`. It's pointing to a new repo `PactSwiftServer` that contains only the versions the `XCFramework` without the history and all the oversized `libpact_ffi.a`s that blow up the repo size.
It will remove Linux support unfortunately, but I'll will work on setting up a separate `PactSwiftLinux` repo soon(ish) where it will be easier to separate and keep support for Swift, Objective-C, Linux and modern swift concurrency separately without nasty hacks and/or interfaces/compiler flags....

**Comment by jasunde** [comment]
Hey, there. I noticed there hasn't been much activity in this repo for a while. I really like Pact testing. What's the status of the project? 
 Are you looking for more contributors?

**Reply by surpher** [reply]
Hi @jasunde. You won't believe this but I did look into the current state of `main` and what it would take to make it releasable. 
I personally haven't touched it in a while due to lack of time on my hands and client engagements that do not do contract testing. I'll try and get it into a stable iOS only state with tests passing and all in the next few weeks. Will revisit the issues and tasks to get improvements in. 

Yes, looking for contributors and help. Always. 👍

**Comment by surpher** [comment]
@ perlguy99 @jasunde 

There has been a substantial refactor of the whole codebase into supporting `async/await` and also moving most of the Pact-building responsibility into `PactSwiftMockServer` and `libpact_ffi` it wraps.

Any help validating the changes and identifying shortfalls would be much appreciated.

branch: `feature/swift-concurrency` :: [link](https://github.com/surpher/PactSwift/tree/feature/swift-concurrency)

**Comment by surpher** [comment]
After a few weeks of thinking about this, the size of the project(s) and maintenance, I've made an executive decision to make destructive changes to the projects.

See https://github.com/surpher/PactSwift/discussions/122

**Comment by surpher** [comment]
@perlguy99 @jasunde 

Hopefully a bit of good news 🤞 

I've spent a few weeks thinking about the approach, and a few days running a few experiments to clean up `PactSwift` and `PactSwiftMockServer`. I think I got to the point where a `v1.2.0` could be released with what's currently on `main` branch. 
It's pretty much the `v1.1.0` with additional changes supporting Swift Concurrency that was introduced on `main`. That means much of the Pact logic is left as is now in `PactSwift` and `PactSwiftMockServer` repos' latest release(s). The branch that moves most of Pact building responsibility into `PactSwiftMockServer` and it's `libpact_ffi.so` is still in WIP state as it is a major refactor job.

The bad news is though, that I will purge the history of all the nonsense that found its way into the repo. Both of them. All the binaries etc in favour of:
- `PactSwift` for darwin, iOS, tvOS:
    - depend on `PactSwiftServer.git` - providing zipped up archive of `PactSwiftMockServer.xcframework` binary. This should only pull down the MBs for that one binary version and not the whole history of `PactSwiftMockServer.git` repo!
- `PactSwift` for Linux:
    - depend on `PactSwiftMockServer.git` with history purged and locked at `v0.4.3` as the new starting point.
    - expose two Swift Package Plugins to help with setup and support automation: 
        - `download-ffi` to download a pre-built `libpact_ffi.so.gz` archive from `https://github.com/pact-foundation/pact-reference/releases`, verifies SHA256 etc... 
        - `libs-dir` to print the path to `libpact_ffi.so` that was downloaded using the `download-ffi` command plugin 
- `PactSwiftMockServer` will converted to a pure swift package (removing Xcode project files and other fluff)
- `PactSwiftMockServer` will trigger release workflow (under certain conditions) and release new binaries atc.

The hard switch will happen in beginning of July.
